std-bits/base/src/main/java/org/enso/base/enso_cloud/EnsoSecretHelper.java

```diff
@@ -70,11 +109,14 @@ public static HttpResponse<InputStream> makeRequest(HttpClient client, Builder b
       }
     }
 
-    // If used a secret then only allow HTTPS
-
     // Build and Send the request.
     var httpRequest = builder.build();
     var bodyHandler = HttpResponse.BodyHandlers.ofInputStream();
-    return client.send(httpRequest, bodyHandler);
+    var javaResponse = client.send(httpRequest, bodyHandler);
+
+    // Extract parts of the response
+    return new EnsoHttpResponse(renderedURI, javaResponse.headers(), javaResponse.body(), javaResponse.statusCode());
   }
+
+  public record EnsoHttpResponse(URI uri, HttpHeaders headers, InputStream body, int statusCode) { }
```

Comment:
I think it would be good to add a comment about what you told my - 'why we wrap the http response into our own `EnsoHttpResponse` type instead of just returning a raw response?' (which is to avoid spilling the secrets in the URI)